components/tidb_query_expr/src/impl_time.rs

```diff
@@ -128,6 +140,40 @@ pub fn week_of_year(ctx: &mut EvalContext, t: Option<&DateTime>) -> Result<Optio
     Ok(Some(Int::from(week)))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn year_week_with_mode(ctx: &mut EvalContext, t: &DateTime, mode: &Int) -> Result<Option<Int>> {
+    if t.invalid_zero() {
+        return ctx
+            .handle_invalid_time_error(Error::incorrect_datetime_value(t))
+            .map(|_| Ok(None))?;
+    }
+
+    let (year, week) = t.year_week(WeekMode::from_bits_truncate(*mode as u32));
+    let result = i64::from(week + year * 100);
```

Comment:
What does `year * 100` mean here?

Reply:
The method implements the `YEARWEEK` builtin function in Mysql: [yearweek](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_yearweek)

```SQL
mysql> SELECT YEARWEEK('1987-01-01');
        -> 198652
```

Here `1986` means the year, and `52` means the 52nd week of the year.

Reply:
Maybe we can add a comment here? Others LGTM.

Reply:
OK

done